app/components/recognition/form.js

```diff
@@ -311,4 +313,87 @@ export default class FormComponent extends Component {
     }
     return awardedBy;
   }
+  @action
+  async handleFileChange(event) {
+    this.currentRecognition.file = event.target.files[0];
+    if (this.currentRecognition.file) {
+      set(this.validationErrors, 'legalResource', null);
+      this.currentRecognition.recognitionModel.legalResource =
+        this.currentRecognition.file.name;
+    }
+  }
+
+  removeFile = task({ drop: true }, async () => {
+    try {
+      if (!this.currentRecognition.file.id) {
+        this.currentRecognition.file = null;
+        return (this.currentRecognition.recognitionModel.legalResource = null);
+      }
+      let response = await fetch(`/files/${this.currentRecognition.file.id}`, {
+        method: 'DELETE',
+      });
+
+      if (response.ok) {
+        this.notify(
+          `Het bestand met de naam ${this.currentRecognition.file.name} is succesvol verwijderd.`,
+          `Bestand succesvol verwijderd.`,
+        );
+        this.currentRecognition.file = null;
+        this.currentRecognition.recognitionModel.legalResource = null;
+        return set(this.validationErrors, 'legalResource', null);
+      }
+      throw new Error('File removal failed');
+    } catch (error) {
+      this.notify(
+        'Er is een fout opgetreden bij het verwijderen van het bestand.',
+        null,
+        'error',
+      );
+      console.error('An error occurred while removing the file', error);
+    }
+  });
+
+  @action
+  async openFileInNewTab(file) {
+    try {
+      if (file) {
+        const url = window.URL.createObjectURL(file);
+        const a = document.createElement('a');
+        a.href = url;
+        a.target = '_blank';
+        document.body.appendChild(a);
+        a.click();
+        a.remove();
+        window.URL.revokeObjectURL(url);
+      }
+    } catch (error) {
+      console.error('An error occurred while opening the file', error);
+    }
+  }
+
+  @action
+  async uploadFile(file) {
+    if (file) {
+      let formData = new FormData();
+      formData.append('file', this.currentRecognition.file);
+      try {
+        let response = await fetch('/files', {
+          method: 'POST',
+          body: formData,
+        });
+
+        if (response.ok) {
+          let {
+            data: { id },
+          } = await response.json();
+          this.currentRecognition.file.id = id;
+          this.currentRecognition.recognitionModel.legalResource = `${id}.pdf`;
```

Comment:
I'm not an expert, but it feels a bit strange to me to only store the file name like this and depend on this being in the `${id}.pdf` format in some places in the code. Wouldn't it be better to actually use a `File` record and add a new relationship to the recognition model?

Reply:
I don't think this was fully resolved yet, but I probably explained it badly. What I meant is that some places in the code assume that the name of the file also contains the id of the resource: `[some-id].pdf`

The code looks up the id by splitting off the `.pdf` suffix and then uses it to do other things. But I think it would be better to add an actual new relationship to the recognition which contains the file record.

So the recognition has either a `legalResource` (which would be a URL), or it has a file relationship (name pending). This would require a data model change, but I think it's a more solid solution imo.

I would discuss this in LBLOD-all maybe, to see what the best solution is for this scenario?

Reply:
@Windvis I checked it with Boris he will provide me the correct relationship and I will implement it like you said. Thanks! 🙂

Reply:
@Windvis thx, this is a valid point.
We can use `http://www.w3.org/2000/01/rdf-schema#seeAlso` to point from the recognition to the file object.